src/couch_replicator_dns.erl

```diff
@@ -0,0 +1,29 @@
+-module(couch_replicator_dns).
+
+-include_lib("kernel/src/inet_dns.hrl").
+-include_lib("ibrowse/include/ibrowse.hrl").
+
+-export([
+    lookup/1
+]).
+
+-spec lookup(iolist()) -> {ok, {term(), pos_integer()}} | {error, term()}.
+
+lookup(Url) ->
+    case ibrowse_lib:parse_url(Url) of
+        #url{host_type = hostname, host = Host} ->
+            case inet_res:resolve(Host, any, any) of
+                {error, {Error, _}} ->
+                    {error, Error};
+                {ok, #dns_rec{anlist=[Answer|_As]}} ->
```

Comment:
Did you inspect that and found dns_rec()? Per docs the return is dns_msg() and opaque, to be accessed using inet_dns: functions, msg() and rr() in this case.